lib/internal/repl.js

```diff
@@ -55,15 +55,26 @@ function createRepl(env, opts, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY !== '') {
```

Comment:
My reasoning is that empty string is a magic value with `setupHistory()`, so it would make more sense to keep it there. Also, if this function ever gets called from anywhere else, the same logic would have to be duplicated.